haas/api.py

```diff
@@ -505,21 +558,65 @@ def headnode_detach_network(headnode, hnic):
                             # Network Code #
                             ################
 
+@rest_call('GET', '/networks')
+def list_networks():
+    get_auth_backend().require_admin()
+
+    networks = db.session.query(model.Network).all()
+    result = {}
+    for n in networks:
+        if n.access:
+            net = {'driver_id': n.network_id, 'projects': [p.label for p in n.access]}
+        else:
+            net = {'driver_id': n.network_id, 'projects': None}
+        result[n.label] = net
+            
+    return json.dumps(result, sort_keys = True)
+
+@rest_call('GET', '/network/<network>/attachments')
+def list_network_attachments(network, project=None):
```

Comment:
So, as written this will try to fish the project arg out of the body of the request, problem:  GET requests don't have bodies.

Did we test this? If it works it's because the libraries aren't being as strict about HTTP semantics as they should be. If not, we should have.

See also my comments in the api docs.